modules/core/src/main/scala/com/azavea/stac4s/ItemCollection.scala

```diff
@@ -1,42 +1,75 @@
 package com.azavea.stac4s
 
 import cats.Eq
+import cats.implicits._
 import io.circe._
 import io.circe.refined._
+import io.circe.syntax._
 
 final case class ItemCollection(
     _type: String = "FeatureCollection",
     stacVersion: StacVersion,
     stacExtensions: List[String],
     features: List[StacItem],
-    links: List[StacLink]
+    links: List[StacLink],
+    extensionFields: JsonObject
 )
 
 object ItemCollection {
 
-  implicit val eqItemCollection: Eq[ItemCollection] = Eq.fromUniversalEquals
-
-  implicit val encItemCollection: Encoder[ItemCollection] = Encoder.forProduct5(
+  val itemCollectionFields = Set(
     "type",
     "stac_version",
     "stac_extensions",
     "features",
     "links"
-  )(itemCollection =>
-    (
-      itemCollection._type,
-      itemCollection.stacVersion,
-      itemCollection.stacExtensions,
-      itemCollection.features,
-      itemCollection.links
-    )
   )
 
-  implicit val decItemCollection: Decoder[ItemCollection] = Decoder.forProduct5(
-    "type",
-    "stac_version",
-    "stac_extensions",
-    "features",
-    "links"
-  )(ItemCollection.apply)
+  implicit val eqItemCollection: Eq[ItemCollection] = Eq.fromUniversalEquals
+
+  implicit val encItemCollection: Encoder[ItemCollection] = new Encoder[ItemCollection] {
+
+    def apply(collection: ItemCollection): Json = {
+      val baseEncoder: Encoder[ItemCollection] = Encoder.forProduct5(
+        "type",
+        "stac_version",
+        "stac_extensions",
+        "features",
+        "links"
+      )(itemCollection =>
+        (
+          itemCollection._type,
+          itemCollection.stacVersion,
+          itemCollection.stacExtensions,
+          itemCollection.features,
+          itemCollection.links
+        )
+      )
+
+      baseEncoder(collection).deepMerge(collection.extensionFields.asJson)
+    }
+  }
+
+  implicit val decItemCollection: Decoder[ItemCollection] = { c: HCursor =>
+    (
+      c.get[String]("type"),
+      c.get[StacVersion]("stac_version"),
+      c.get[Option[List[String]]]("stac_extensions"),
+      c.get[List[StacItem]]("features"),
+      c.get[Option[List[StacLink]]]("links"),
+      c.value.as[JsonObject]
```

Comment:
Here's the second part of the big increase in verbosity -- derivation is no longer correct for any of these, since we need "root document without the vanilla fields"